AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -215,39 +200,26 @@ class CompleteScreenViewModel: ObservableObject {
         return
     }
     
-    private func getMeasurementsAndDisplayData() {
-        let streams = session.stream.map(\.id)
-        
-        service.downloadMeasurements(streamsIds: streams) { [weak self] result in
-            guard let self = self else { return }
-            switch result {
-            case .failure(let error):
-                Log.error("Failed to download session: \(error)")
-                self.showAlert()
-            case .success(let downloadedStreamsWithMeasurements):
-                guard !downloadedStreamsWithMeasurements.isEmpty else {
-                    Log.error("Session has no streams")
-                    self.showAlert()
-                    return
-                }
-                
-                DispatchQueue.main.async {
-                    self.externalSessionWithStreams = self.service.createExternalSession(from: self.session, with: downloadedStreamsWithMeasurements)
-                    
-                    self.sessionStreams = .ready( self.externalSessionWithStreams!.streams.map {
-                        .init(id: $0.id,
-                              sensorName: Self.getSensorName($0.sensorName),
-                              sensorUnit: $0.unitSymbol,
-                              lastMeasurementValue: $0.measurements.last?.value ?? 0,
-                              color: $0.thresholdsValues.colorFor(value: $0.measurements.last?.value ?? 0),
-                              measurements: $0.measurements.map({.init(value: $0.value, time: $0.time, latitude: $0.latitude, longitude: $0.longitude)}), thresholds: $0.thresholdsValues)
-                    })
-                    
-                    if let stream = self.externalSessionWithStreams!.streams.first {
-                        self.assignValues(with: stream)
-                        self.defineChartRange(with: stream)
-                    }
-                }
+    private func createExternalSession(with sortedStreams: [MeasurementsDownloaderResultModel.Stream]) {
+        DispatchQueue.main.async {
+            self.externalSessionWithStreams = self.service.createExternalSession(from: self.session, with: sortedStreams)
+            
+            self.sessionStreams = .ready(self.externalSessionWithStreams!.streams.map {
+                .init(id: $0.id,
+                      sensorName: Self.getSensorName($0.sensorName),
+                      sensorUnit: $0.unitSymbol,
+                      lastMeasurementValue: $0.measurements.last?.value ?? 0,
```

Comment:
I know this code was already here but I think we should add some guard that measurements are not empty and if they are we should indicate that in the view instead of showing 0. Maybe it would be a good idea to create a ticket for that?